src/sdk/config_fs.js

```diff
@@ -911,6 +970,20 @@ class ConfigFS {
         }
         return config_file_names;
     }
+
+    /**
+     * _throw_if_config_dir_locked validates that
+     * config dir schema version on system.json matches the config_dir_version on package.json
+     * throws an error if they do not match.
+     * @returns {Promise<void>}
+     */
+    async _throw_if_config_dir_locked() {
+        const system_data = await this.get_system_config_file({silent_if_missing: true});
+        if (!system_data) return;
+        if (this.config_dir_version !== system_data.config_directory.config_dir_version) {
```

Comment:
This check is also related to the meaning of the `config_dir_version`

Reply:
used version_compare, is this what you meant?